packages/smithy-client/src/extensions/defaultExtensionConfiguration.ts

```diff
@@ -34,5 +36,6 @@ export const getDefaultClientConfiguration = getDefaultExtensionConfiguration;
 export const resolveDefaultRuntimeConfig = (config: DefaultExtensionConfiguration) => {
   return {
     ...resolveChecksumRuntimeConfig(config),
-  };
+    ...resolveRetryRuntimeConfig(config),
+  } as RuntimeConfigType;
```

Comment:
nit: would prefer the return type to be in the function signature rather than using `as`.